pyaerocom/io/pyaro/read_pyaro.py

```diff
@@ -0,0 +1,292 @@
+                    msg = f"TS type {ts_type} of station {current_station} is different from already found value {ts_types[current_station]}"
+                    logger.error(msg)
+                    raise ValueError(msg)
+
+                data_array[idx, :] = ungriddeddata_line
+
+                #  Fills meta_idx
+                meta_idx[station_idx[current_station]][var].append(idx)
+
+                idx += 1
+
+        new_meta_idx = {}
+        for station_id in meta_idx:
+            new_meta_idx[station_id] = {}
+            for var_id in meta_idx[station_id]:
+                new_meta_idx[station_id][var_id] = np.array(meta_idx[station_id][var_id])
+
+        self.data._data = data_array
+        self.data.meta_idx = new_meta_idx
+        self.data.metadata = self._add_ts_type_to_metadata(metadata, ts_types)
+        self.data.var_idx = var_idx
+
+        return self.data
+
+    def _make_ungridded_metadata(
+        self, stations: dict[str, Station], var_idx: dict[str, int], units: dict[str, str]
+    ) -> Metadata:
+        idx = 0
+        metadata = {}
+        for name, station in stations.items():
+            metadata[idx] = dict(
+                data_id=self.config.name,
+                variables=list(self.get_variables()),
+                var_info=units,
+                instrument_name="",
+                latitude=station["latitude"],
+                longitude=station["longitude"],
+                altitude=station["altitude"],
+                station_name=name,
+                station_id=name,
+                country=station["country"],
+                ts_type="undefined",  # TEMP
+                data_revision="n/d",  # Temp: Need to be changed. Must add way of getting this from Reader
+            )
+            idx += 1
+
+        return Metadata(metadata)
+
+    def _pyaro_dataline_to_ungriddeddata_dataline(
+        self, data: np.void, idx: int, var_idx: int
+    ) -> np.ndarray:
+        new_data = np.zeros(12)
+        new_data[self._METADATAKEYINDEX] = idx
+        new_data[self._TIMEINDEX] = data["start_times"]
+        new_data[self._LATINDEX] = data["latitudes"]
+        new_data[self._LONINDEX] = data["longitudes"]
+        new_data[self._ALTITUDEINDEX] = data["altitudes"]
+        new_data[self._VARINDEX] = var_idx
+        new_data[self._DATAINDEX] = data["values"]
+        new_data[self._DATAHEIGHTINDEX] = np.nan
+        new_data[self._DATAERRINDEX] = data["standard_deviations"]
+        new_data[self._DATAFLAGINDEX] = data["flags"]
+        new_data[self._STOPTIMEINDEX] = data["end_times"]
+        new_data[self._TRASHINDEX] = np.nan
+
+        return new_data
+
+    def _calculate_ts_type(self, start: np.datetime64, stop: np.datetime64) -> TsType:
+        seconds = (stop - start).astype("timedelta64[s]").astype(np.int32)
+        if seconds == 0:
+            ts_type = TsType(
+                "daily"
+            )  # TODO this should be instentanious, but that tstype does not exist
```

Comment:
Maybe hourly?

Reply:
Changed to hourly. Still just a quick fix though...